File: SORTING/SELECTION-SORT/C/selectionsort.c
//SELECTION SORT
//USING MIN ELEMENT
//ASCENDING
#include <stdio.h>
void selectionsort(int arr[],int n)
{
  int min=0;
  for(register int i=0;i<n;i++)
  {
    min=i;
    for(register int j=i;j<n;j++)
    {
      if(arr[min]>arr[j])
      {
        min=j;
      }
    }
    //swap
    int temp = arr[i];
    arr[i] =arr[min];
    arr[min]=temp;
  }
}
int main()
{
  printf("Enter the size of the array\n");
  int n;
  scanf("%d",&n);
  int arr[n];
  printf("Enter the elements of the array\n");
  for(register int i=0;i<n;i++)
  {
    scanf("%d",&arr[i]);
  }
  selectionsort(arr,n);
  printf("Sorted array:\n");
  for(register int i=0;i<n;i++)
  {
    printf("%d ",arr[i]);
  }
  printf("\n");
}

File: SORTING/SELECTION-SORT/JAVA/selectionsort.java
//descending
//using min element
import java.util.*;

public class selectionsort
{
    public static void main(String[] args)
    {
        Scanner in = new Scanner(System.in); //scanner for input
        int n; //size of the array
        System.out.println("Enter the size of the array");
        n = in.nextInt();
        int[] arr = new int[n]; //declared array
        System.out.println("Enter the elements");
        for(int i=0;i<n;i++) //input array
        {
            arr[i] = in.nextInt(); 
        }
        selection(arr);
        in.close(); 
        for(int k=0;k<arr.length;k++)
        {
            System.out.print(arr[k]+" ");
        }
        System.out.println();
    }
    public static void selection(int[] array)
    {
        int i,j;
        int min=0; //minimum index
        for(i=array.length-1;i>0;i--)
        {
            min=0; //set minimum index to 0
            for(j=0;j<=i;j++) //finding min index in the elements
            {
                if(array[min] > array[j])
                {
                    min = j;
                }
            }
            swap(array,min,i);
        }
    }
    public static void swap(int[] arr,int a,int b)
    {
        //swapping
        int t = arr[a];
        arr[a] = arr[b];
        arr[b] = t;
    }
}